lib/dry/container/mixin.rb

```diff
@@ -105,6 +105,8 @@ def register(key, contents = nil, options = EMPTY_HASH, &block)
         config.registry.call(_container, key, item, options)
 
         self
+      rescue FrozenError
+        raise FrozenError, "can't modify frozen Dry::Container (when attempting to register '#{key}')"
```

Comment:
```suggestion
        raise FrozenError, "can't modify frozen #{self.class} (when attempting to register '#{key}')"
```

Reply:
I actually tried this originally but it's very tricky to test with an anonymous (unnamed) class. 

Changing _just_ this line causes three test failures, which are simple enough to work around individually, but hard to work around in a way that works for all of them (since this is a shared example, it has to work for all of them).

1. Mixin extended: `FrozenError: can't modify frozen Class (when attempting to register 'baz')`
2. Mixin included `FrozenError: can't modify frozen #<Class:0x00007fc6af9f5ec0> (when attempting to register 'baz')`
3. Class `FrozenError: can't modify frozen Dry::Container (when attempting to register 'baz')`

I tried redefining the `klass` for this test but that won't work because the klass is contructed in different ways for each. And defining the `klass` with `class MyTestClass; include Dry::Container::Mixin; end;` doesn't work because then we're redefining the constant, and, yanno, we freeze the klass in these tests. We could `Object.remove_const` but I think we want go down that path for such a small feature.

Went with a simpler, though less robust method, of using a regex to match the message is right _except_ for the actual class name, since there are so many valid options there. I went for `\S+` (one or more non-whitespace characters). I think that should work? Don't think `self.class` will ever return something with a space in it, but I could be wrong. Can change that to some other regex metacharacter.